can_validator/fsae_can_spec.py

```diff
@@ -0,0 +1,38 @@
+"""
+Generate fsae_can_spec.h file.
+Run this file to write just fsae_can_spec.h or main.py to write all files.
+"""
+import sys
+sys.path.append("ParseCAN")
+import ParseCAN
+from common import fsae_can_spec_path, spec_path
+import re
+
+
+def write(output_path, spec_path):
+    """
+    Generate fsae_can_spec.h file.
+
+    :param output_path: file to be written to
+    :param spec_path: CAN spec path
+    """
+    spec = ParseCAN.spec.can(spec_path)
+    with open(output_path, 'w') as f:
+        clean_output_path = re.sub('[^A-Za-z0-9_]+', '_', output_path).upper()
+        f.write('#ifndef ____' + clean_output_path + '\n')
+        f.write('#define ____' + clean_output_path + '\n\n')
+        for id, message in spec.messages.items():
+            f.write('#define ' + message.name.upper() + '__id ' + str(id) + '\n')
+            for segment_name, segment in message.segments.items():
```

Comment:
I did not intend for a developer to know that message.segments is a dict, especially since I want it to be at my discretion to map by names or by another property or properties. Would you find it less useful if you got an iterable with messages in it? Do you need fast searching by name?

Reply:
Same goes for messages and IDs. In fact you'd be getting totally confusing results with the latest ParseCAN, because I map doubly by name and by id, iirc.

Reply:
I think an iterable with messages would be fine (fast searching by name is not really important). How did you want me to iterate through message.segments if not by calling items?

Reply:
I will expose iterables and completely hide the dictionaries. These will be synonymous to the dictionaries as they are now (msg.segments will be an iterable). You'll still be able to search by name, although you might not be able to use bracket notation (msg.get_segment_by_name(name) instead of msg.segments[name]).